internal/jennies/golang/templates/types/struct_equality_method.tmpl

```diff
@@ -0,0 +1,92 @@
+func (resource {{ .def.Name|upperCamelCase }}) Equals(other {{ .def.Name|upperCamelCase }}) bool {
+	{{- template "type_equality_check" (dict "Type" .def.Type "Nullable" .def.Type.Nullable "Dereference" false "SelfName" "resource" "OtherName" "other" "Depth" 0) }}
+
+	return true
+}
+
+{{/* arguments: Type, Nullable, Dereference, SelfName, OtherName */}}
+{{- define "type_equality_check" }}
+	{{- if .Type.IsAny }}
+		{{- $dereference := ternary "*" "" .Dereference }}
+		// is DeepEqual good enough here?
+		if !reflect.DeepEqual({{ $dereference }}{{ .SelfName }}, {{ $dereference }}{{ .OtherName }}) {
+			return false
+		}
+	{{- else if resolvesToArray .Type }}
+		{{- $needsDereference := and .Type.IsRef .Type.Nullable }}
+		{{- $dereference := ternary "*" "" $needsDereference }}
+		{{- $depth := default 1  .Depth }}
+		{{- if $needsDereference }}
+			if !(({{ .SelfName }} == nil && {{ .OtherName }} == nil) || ({{ .SelfName }} != nil && {{ .OtherName }} != nil)) {
```

Comment:
Isn't the same than: 
```
{{ .SelfName }} == nil && {{ .OtherName }} != nil || {{ .SelfName }} != nil && {{ .OtherName }} == nil
```

You can remove some parentheses. And same for the one in map.